log-analyzer/features/org.wso2.carbon.la.apim.feature/src/main/resources/gadgets/LogAPIMArtifactDeleted/js/VizGrammar.js

```diff
@@ -0,0 +1,2300 @@
+    this.metadata = dataTable[0].metadata;
+    var marks =[];
+    var signals = [];
+    this.spec = {};
+
+    config = checkConfig(config, this.metadata);
+    this.config = config;
+    dataTable[0].name= config.title;
+
+    var xScale = {
+        "name": "x",
+        "type": this.metadata.types[config.x],
+        "range": "width",
+        "zero": config.zero,
+        "domain": {"data":  config.title, "field": this.metadata.names[config.x]}
+    };
+
+    var yScale = {
+        "name": "y",
+        "type": this.metadata.types[config.y],
+        "range": "height",
+        "zero": config.zero,
+        "domain": {"data":  config.title, "field": this.metadata.names[config.y]}
+    };
+
+    var scales =  [xScale, yScale];
+
+    if (config.color != -1) {
+
+        if (config.colorDomain == -1) {
+            config.colorDomain = {"data":  config.title, "field": this.metadata.names[config.color]};
+        }
+
+        var colorScale = {
+            "name": "color",
+            "type": "ordinal",
+            "domain": config.colorDomain,
+            "range": config.colorScale
+        };
+        scales.push(colorScale);
+    }
+
+    var axes =  getXYAxes(config, "x", "x", "y", "y");
+
+    marks.push(getLineMark(config, this.metadata));
+    config.markSize = 20;
+    marks.push(getSymbolMark(config, this.metadata));
+
+    if (config.range) {
+        signals = getRangeSignals(config, signals);
+        marks = getRangeMark(config, marks);
+    }
+
+    if (config.color != -1) {
+
+        var legendTitle = "Legend";
+
+        if (config.title != "table") {
+            legendTitle = config.title;
+        }
+
+        var legends = [
+            {
+                "fill": "color",
+                "title": "Legend",
+                "offset": 0,
+                "properties": {
+                    "symbols": {
+                        "stroke": {"value": "transparent"}
+                    },
+                    "title": {
+                        "fill": {"value": config.legendTitleColor},
+                        "fontSize": {"value": config.legendTitleFontSize}
+                    },
+                    "labels": {
+                        "fill": {"value": config.legendTextColor},
+                        "fontSize": {"value": config.ledgendTextFontSize}
+                    }
+                }
+            }
+        ];
+
+        this.spec.legends = legends;
+    }
+
+    this.spec.width = config.width;
+    this.spec.height = config.height;
+    this.spec.axes = axes;
+    this.spec.data = dataTable;
+    this.spec.scales = scales;
+    this.spec.padding = config.padding;
+    this.spec.marks = marks;
+    this.spec.signals = signals;
+
+};
+
+line.prototype.draw = function(div, callbacks) {
+    if(this.config.maxLength != -1){
+        var dataset = this.spec.data[0].values;
+        var maxValue = this.config.maxLength;
+        if(dataset.length >= this.config.maxLength){
```

Comment:
Hi Chandana,
We have replace the Vizgrammer.js with Vizgrammer.min.js and relevant commits are in

https://github.com/wso2/shared-analytics/pull/126/files#diff-6ac780d61b2b6d470c8ff04bdeed3bcdR52

Thanks,
SajithD